exercises/practice/anagram/anagram.pony

```diff
@@ -0,0 +1,6 @@
+primitive Anagram
+  fun apply(base: String, anagrams: Array[String]): Array[String] =>
+    """
+    Returns all candidates that are anagrams of, but not equal to, 'base'.
```

Comment:
We might also rename `base` to `word` and document it as "not equal to the original word" rather than use the word "base" here.  The original word is what the anagrams are based on, but probably clearer to reference it as the word it is.

Reply:
I'd like to pretend I did all the work but I "borrowed"™ a lot of the documentation comments from the elixir track :-)

Done!